### PR TITLE
Features/out 79 realtime update

### PR DESCRIPTION
**Pull request by Dat-H-Tran**
Implement the following 
[complete  OUT-82](https://linear.app/pdw/issue/OUT-82/sidebar-ui-updates)
[complete OUT-81](https://linear.app/pdw/issue/OUT-81/setup-github-action-to-deploy-fe)
[complete  OUT-70](https://linear.app/pdw/issue/OUT-70/detect-messages-that-twilio-failed-to-deliver)

### CODE REVIEW

File: src/components/broadcaster/LastBroadcastStatus.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        totalSecondSent: data.data.past[0]?.totalSecondSent,
+        successfullyDelivered: data.data.past[0]?.successfullyDelivered,
+        failedDelivered: data.data.past[0]?.failedDelivered
+      }
+      setRenderMostRecentBroadcastDetails({
+        totalFirstSent: newMostRecent.totalFirstSent ?? 0,
+        totalSecondSent: newMostRecent.totalSecondSent ?? 0,
+        successfullyDelivered: newMostRecent.successfullyDelivered ?? 0,
+        failedDelivered: newMostRecent.failedDelivered ?? 0
+      })
```
Comment (kyphan1) [inline src/components/broadcaster/LastBroadcastStatus.tsx:38]:
```suggestion
      const newMostRecent = {
        totalFirstSent: data.data.past[0]?.totalFirstSent ?? 0,
        totalSecondSent: data.data.past[0]?.totalSecondSent ?? 0,
        successfullyDelivered: data.data.past[0]?.successfullyDelivered ?? 0,
        failedDelivered: data.data.past[0]?.failedDelivered ?? 0
      }
      setRenderMostRecentBroadcastDetails({
        ...newMostRecent
      })
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
+name: Build and Deploy
+on:
+  push:
+#    branches:
+#      - main
```
Comment (kyphan1) [inline .github/workflows/deploy.yml:5]:
```suggestion
```

File: src/components/broadcaster/BroadcastDashBoard.tsx [hunk 3]
```diff
@@ -17,49 +21,44 @@ const BroadcastDashboard = () => {
     setIsFirstMessage(isFirst)
   }
 
+  const handleSendNow = () => {
+    void makeBroadcast()
+  }
```
Comment (kyphan1) [inline src/components/broadcaster/BroadcastDashBoard.tsx:26]: We should create a react-query useMutation. That way you could access `pending` state.

File: src/components/broadcaster/LastBroadcastStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        failedDelivered: initialData.pages[0].data.past[0]?.failedDelivered,
+        totalUnsubscribed: initialData.pages[0].data.past[0]?.totalUnsubscribed
+      }
+      setRenderMostRecentBroadcastDetails({
+        totalFirstSent: newMostRecent.totalFirstSent ?? 0,
+        totalSecondSent: newMostRecent.totalSecondSent ?? 0,
+        successfullyDelivered: newMostRecent.successfullyDelivered ?? 0,
+        failedDelivered: newMostRecent.failedDelivered ?? 0,
+        totalUnsubscribed: newMostRecent.totalUnsubscribed ?? 0
+      })
```
Comment (kyphan1) [inline src/components/broadcaster/LastBroadcastStatus.tsx:40]:
```suggestion
      const newMostRecent = {
        totalFirstSent: initialData.pages[0].data.past[0]?.totalFirstSent ?? 0,
        totalSecondSent: initialData.pages[0].data.past[0]?.totalSecondSent ?? 0,
        successfullyDelivered: initialData.pages[0].data.past[0]?.successfullyDelivered ?? 0,
        failedDelivered: initialData.pages[0].data.past[0]?.failedDelivered ?? 0,
        totalUnsubscribed: initialData.pages[0].data.past[0]?.totalUnsubscribed ?? 0
      }
      setRenderMostRecentBroadcastDetails(newMostRecent)
```

File: src/components/broadcaster/LastBroadcastStatus.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  const { data, isPending } = useBroadcastDashboardQuery(queryClient)
+
+  const [renderMostRecentBroadcastDetails, setRenderMostRecentBroadcastDetails] = useState<BroadcastSentDetail>({
+    failedDelivered: 0,
+    successfullyDelivered: 0,
+    totalFirstSent: 0,
+    totalSecondSent: 0
+  })
+  const mostRecentBroadcastDetails = useSubscribeMostRecentBroadcastDetail() // may has one or all fields as undefined/no data
+  const { data: pastData } = usePastBroadcastsQuery(initialData)
```
[the remark this answers is not in the document]
Comment (Dat-H-Tran) [reply src/components/broadcaster/LastBroadcastStatus.tsx:22]: This is planned to be redone in a latter PR

File: src/components/broadcaster/BroadcastDashBoard.tsx (hunk 3 above)
Comment (Dat-H-Tran) [reply src/components/broadcaster/BroadcastDashBoard.tsx:26]: We don't need to as we switched to using missive's button

**Review by giahung68 — APPROVED** [review]

**Pull request merged**